packages/manager/src/components/EnhancedSelect/Select.tsx

```diff
@@ -67,22 +67,54 @@ export interface BaseSelectProps<
   Clearable extends boolean = false
 > extends Omit<SelectProps<I, IsMulti>, 'onChange'>,
     CreatableSelectProps<I, IsMulti> {
+  /**
+   * Optional additional class names to apply to the root element.
+   */
   classes?: any;
+  /**
+   * Allow options to be created while the isLoading prop is true.
+   * @default false
+   */
   creatable?: boolean;
+
   /** the rest are props we've added ourselves */
+  /**
+   * If true, the input will be blurred when a selection is made.
+   * @default false
+   */
   disabled?: boolean;
+  /**
+   * To facilitate error handling in forms.
+   */
   errorGroup?: string;
-
   /**
-   * errorText and label both passed to textFieldProps
-   * @todo consider just putting this under textFieldProps
+   * errorText and label both passed to textFieldProps.
+   * @todo consider just putting this under textFieldProps.
    */
   errorText?: string;
-
+  /**
+   * Returns helper text to display under the select popover.
+   */
   guidance?: React.ReactNode | string;
+  /**
+   * If true, the label will be hidden.
+   * @default false
+   */
   hideLabel?: boolean;
+  /**
+   * If true, the label will be displayed inline.
+   * @default false
+   */
   inline?: boolean;
+  /**
+   * To specify a custom input ID
+   * @default false
+   */
   inputId?: any;
```

Comment:
Optional: replace `any` with `string`. (This causes one other type error that would have to be addressed)

```suggestion
  inputId?: string;
```

Reply:
cool, typed this better